### PR TITLE
Upgrade pandas supported range to >=2.0, <2.3

### PR DESCRIPTION
**Pull request by shchur**
*Issue #, if available:* #3989

*Description of changes:*
- Raise pandas upper bound to 2.2 (inclusive)
- Update time series pandas-related functionality & tests to work with the [new pandas offset aliases](https://pandas.pydata.org/docs/whatsnew/v2.2.0.html#deprecate-aliases-m-q-y-etc-in-favour-of-me-qe-ye-etc-for-offsets) (breaking change in pandas 2.2)
  - To ensure compatibility with both pandas 2.0/2.1 (old aliases) and 2.2 (new aliases), we adopt the following strategy:
     - All aliases are mapped to the new version (e.g., we always use `YE` instead of `A` in our code).
     - For libraries such as GluonTS that don't support pandas 2.2, we replace all aliases with dummy values (`W`) that are compatible with both old/new style, and take care of all the frequency-related logic in AutoGluon (e.g., time features, lag selection)
     - We cap GluonTS dependency to `>=0.14.0,<0.14.4` to ensure that pandas 2.2 can be installed
   - Most changes in this PR are related to the tests, ensuring that they can run with both 2.0, 2.1 and 2.2.
- Fix warning in `CategoryFeatureGenerator`

*Testing:*
- Our platform tests currently cover pandas 2.0 (used in all Py3.8 tests) and pandas 2.2 (all remaining tests). We also did a run with pandas 2.1 (https://github.com/autogluon/autogluon/actions/runs/8636868243), so we should be safe to merge.

*To do:*
- [ ]  @Innixma Verify status of Tabular support
- [ ] @zhiqiangdon Verify status of AutoMM support
- [x] @shchur Verify status of TimeSeries support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| idna==3.6 | idna==3.7 |
| idna==3.6 | idna==3.7 |

File: features/src/autogluon/features/generators/drop_duplicates.py [hunk 1]
```diff
@@ -132,6 +132,7 @@ def _drop_duplicate_features_categorical(cls, X: DataFrame, keep: Union[str, boo
             # Converts [5, 'a', np.nan, 5] to [0, 1, 2, 0], these would be considered duplicates since they carry the same information.
 
             # Have to convert to object dtype because category dtype for unknown reasons will refuse to replace NaNs.
+            # TODO: Fix FutureWarning
             X_cur = X[features_to_check].astype("object").replace(mapping_features_val_dict_cur).astype(np.int64)
```
Comment (shchur) [inline features/src/autogluon/features/generators/drop_duplicates.py:136]: @Innixma this line raises `FutureWarning`:
```
  /local/home/shchuro/workspace/autogluon/features/src/autogluon/features/generators/drop_duplicates.py:135: FutureWarning: Downcasting behavior in `r
eplace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-
in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
    X_cur = X[features_to_check].astype("object").replace(mapping_features_val_dict_cur).astype(np.int64)
```
Do you have any recommendations on how to fix it?

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| gluonts==0.14.3 | gluonts==0.14.4 |
| pandas==2.2.2 | pandas==2.1.4 |
| gluonts==0.14.3 | gluonts==0.14.4 |
| pandas==2.2.2 | pandas==2.1.4 |

File: timeseries/src/autogluon/timeseries/utils/datetime/base.py [hunk 2]
```diff
@@ -1,39 +1,57 @@
 import pandas as pd
 
 TO_MAJOR_FREQ = {
-    "min": "T",
-    "ms": "L",
-    "us": "U",
+    # sub-daily
```
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/utils/datetime/base.py:4]: wow...

**Review by canerturkmen — COMMENTED** [review]
LGTM! Thanks!

**Review by zhiqiangdon — COMMENTED** [review]
May not have enough time to test this before 1.1 release.

**Review by Innixma — APPROVED** [review]
LGTM! Will do follow-up PRs to addres FutureWarnings

File: features/src/autogluon/features/generators/drop_duplicates.py (hunk 1 above)
Comment (Innixma) [reply features/src/autogluon/features/generators/drop_duplicates.py:136]: will investigate and send PR

**Pull request merged**